### ISSUE TITLE
Error when using `TargetEncoder` within `DataFrameMapper` within pipeline

### ISSUE THREAD

**Issue by woodly0** [jpmml/sklearn2pmml#388]
Hello Sir,

I'm starting to getting a hang of your package. Really good work!
Again, I'd be glad if you could help me. Let's first create some random data:

```py
import pandas as pd
from category_encoders import TargetEncoder
from sklearn.linear_model import LogisticRegression
from sklearn.preprocessing import OrdinalEncoder
from sklearn.datasets import make_classification
from sklearn_pandas import DataFrameMapper
from sklearn2pmml import sklearn2pmml
from sklearn2pmml.preprocessing import ExpressionTransformer
from sklearn2pmml.pipeline import PMMLPipeline
from sklearn2pmml.decoration import Alias

X, y = make_classification(
    n_samples=1000,
    n_features=2,
    n_classes=2,
    n_redundant=0,
    n_informative=2,
    random_state=1,
)

X = pd.DataFrame(X, columns=["feature1", "feature2"]).applymap(
    lambda x: "RED" if x < 0 else "BLUE"
)
y = pd.Series(y)
```
Then I create a mapper that encodes the 2 categorical features:

```py
mapper = DataFrameMapper(
    [
        (
            ["feature1"],
            [
                Alias(ExpressionTransformer("X[0].lower()"), "feature1" + "_lower"),
                OrdinalEncoder(
                    categories=[
                        ["red", "blue", "green"],
                    ],
                    dtype=int,
                    handle_unknown="error",
                    encoded_missing_value=0,
                ),
            ],
        ),
        (
            ["feature2"],
            [
                TargetEncoder(
                    return_df=False,
                    min_samples_leaf=49,
                    smoothing=1,
                ),
            ],
        ),
    ],
    df_out=True,
    default=False,
)
```
This works very well when tested on its own. Then I create and fit a `PMMLPipeline`:

```py
pmml_pipe = PMMLPipeline(
    [
        ("mapper", mapper),
        ("classifier", LogisticRegression(solver="liblinear", class_weight="balanced")),
    ]
)
pmml_pipe.fit(X, y)
```
which works still fine. However, when trying to export it:

```py
sklearn2pmml(pmml_pipe, "out.pmml", with_repr=True)
```
which then throws this error that I don't understand:

```
Standard output is empty
Standard error:
Exception in thread "main" java.lang.NullPointerException: Cannot invoke "pandas.core.Series.getBlockManager()" because "series" is null
	at pandas.core.SeriesUtil.toMap(SeriesUtil.java:37)
	at category_encoders.OrdinalMapEncoder.encodeFeatures(OrdinalMapEncoder.java:106)
	at sklearn.Transformer.encode(Transformer.java:70)
	at sklearn_pandas.DataFrameMapper.encodeFeatures(DataFrameMapper.java:67)
	at sklearn.Transformer.encode(Transformer.java:70)
	at sklearn.Composite.encodeFeatures(Composite.java:119)
	at sklearn2pmml.pipeline.PMMLPipeline.encodePMML(PMMLPipeline.java:154)
	at com.sklearn2pmml.Main.run(Main.java:91)
	at com.sklearn2pmml.Main.main(Main.java:66)
```
What does that mean?

**Comment by vruusmann** [comment]
> Exception in thread "main" java.lang.NullPointerException: Cannot invoke "pandas.core.Series.getBlockManager()" because "series" is null
>	at pandas.core.SeriesUtil.toMap(SeriesUtil.java:37)
>

First things first - this error stack trace is not in sync with recent SkLearn2PMML package versions.

Please upgrade your SkLearn2PMML package to latest (should be 0.94.0 right now). Does the error persist after the upgrade?

If the error persists, please post an updated error stack trace.

>  What does that mean?

It basically means that the pickled `pandas.Series` object is missing some critical attribute.

The SkLearn2PMML package should work with Pandas 1.5.X and 2.0.X versions. If you're using any other Pandas version, then please upgrade it as well!

**Comment by woodly0** [comment]
Thanks for your quick answer. 
> Please upgrade your SkLearn2PMML package to latest (should be 0.94.0 right now). Does the error persist after the upgrade?

The update is done. The new stack trace is the following:

```
Standard output is empty
Standard error:
Exception in thread "main" java.lang.NullPointerException: Cannot invoke "pandas.core.Series.getBlockManager()" because "series" is null
	at pandas.core.SeriesUtil.toMap(SeriesUtil.java:67)
	at category_encoders.OrdinalMapEncoder.encodeFeatures(OrdinalMapEncoder.java:106)
	at sklearn.Transformer.encode(Transformer.java:72)
	at sklearn_pandas.DataFrameMapper.encodeFeatures(DataFrameMapper.java:67)
	at sklearn.Transformer.encode(Transformer.java:72)
	at sklearn.Composite.encodeFeatures(Composite.java:119)
	at sklearn2pmml.pipeline.PMMLPipeline.encodePMML(PMMLPipeline.java:152)
	at com.sklearn2pmml.Main.run(Main.java:91)
	at com.sklearn2pmml.Main.main(Main.java:66)
```

- pandas v 2.0.1
- scikit-learn v 1.2.2

**Comment by woodly0** [comment]
> Thanks for posting a working code example (including the dataset!), which allows me to reproduce the problem on my own computer

With pleasure! That's the least I can do.

> On this line, replace Object col = cols.get(i) with Object col = ScalarUtil.decode(cols.get(i)), and everything works OK.
Will add this fix into the next SkLearn2PMML release (about a week from today?).

That is so nice. Thank you very much!

**Comment by vruusmann** [comment]
The fix is included in SkLearn2PMML 0.94.1 (and newer).
